server/grpc_service.go

```diff
@@ -519,13 +520,17 @@ func (s *GrpcServer) PutStore(ctx context.Context, request *pdpb.PutStoreRequest
 	log.Info("put store ok", zap.Stringer("store", store))
 	CheckPDVersion(s.persistOptions)
 	if !core.IsStoreContainLabel(request.GetStore(), core.EngineKey, core.EngineTiFlash) {
-		go func(url string) {
+		go func(ctx context.Context, url string) {
+			select {
 			// tikv may not ready to server.
-			time.Sleep(5 * time.Second)
-			if err := s.storeConfigManager.Load(url); err != nil {
-				log.Error("load store config failed", zap.String("url", url), zap.Error(err))
+			case <-time.After(storeReadyWaitTime):
+				if err := s.storeConfigManager.Load(url); err != nil {
+					log.Error("load store config failed", zap.String("url", url), zap.Error(err))
+				}
+			case <-ctx.Done():
+				return
 			}
-		}(store.GetStatusAddress())
+		}(s.Server.Context(), store.GetStatusAddress())
```

Comment:
only the leader needs to sync or every PD needs to?

Reply:
In past, the configuration of the stores is used by merge-checker and some monitor, so only the pd leader should know the configuration

Reply:
After the leader is changed, it won't exit this goroutine?

Reply:
yes, I will use serverLoopCtx.